platform-sdk/swirlds-common/src/test/java/com/swirlds/common/wiring/schedulers/SequentialTaskSchedulerTests.java

```diff
@@ -0,0 +1,1966 @@
+                .build()
+                .cast();
+
+        final InputWire<Integer, Integer> inputA = taskSchedulerA.buildInputWire("inputA");
+        final InputWire<Integer, Integer> inputB = taskSchedulerB.buildInputWire("inputB");
+        final InputWire<Integer, Integer> inputC = taskSchedulerC.buildInputWire("inputC");
+        final InputWire<Integer, Void> inputD = taskSchedulerD.buildInputWire("inputD");
+
+        taskSchedulerA.solderTo(inputB);
+        taskSchedulerB.solderTo(inputC);
+        taskSchedulerC.solderTo(inputD);
+
+        final AtomicInteger countA = new AtomicInteger();
+        final AtomicInteger countB = new AtomicInteger();
+        final AtomicInteger countC = new AtomicInteger();
+        final AtomicInteger countD = new AtomicInteger();
+
+        inputA.bind(x -> {
+            countA.set(hash32(countA.get(), x));
+            return x;
+        });
+
+        inputB.bind(x -> {
+            countB.set(hash32(countB.get(), x));
+            return x;
+        });
+
+        inputC.bind(x -> {
+            countC.set(hash32(countC.get(), x));
+            return x;
+        });
+
+        inputD.bind(x -> {
+            countD.set(hash32(countD.get(), x));
+        });
+
+        int expectedCount = 0;
+
+        for (int i = 0; i < 100; i++) {
+            inputA.put(i);
+            expectedCount = hash32(expectedCount, i);
+        }
+
+        assertEventuallyEquals(
+                expectedCount, countD::get, Duration.ofSeconds(1), "Wire sum did not match expected sum");
+        assertEquals(expectedCount, countA.get());
+        assertEquals(expectedCount, countB.get());
+        assertEquals(expectedCount, countC.get());
+    }
+
+    @Test
+    void multipleOutputChannelsTest() {
+        final TaskScheduler<Integer> taskSchedulerA =
+                model.schedulerBuilder("A").build().cast();
+        final InputWire<Integer, Integer> aIn = taskSchedulerA.buildInputWire("aIn");
+        final SecondaryOutputWire<Boolean> aOutBoolean = taskSchedulerA.buildSecondaryOutputWire();
+        final SecondaryOutputWire<String> aOutString = taskSchedulerA.buildSecondaryOutputWire();
+
+        final TaskScheduler<Void> taskSchedulerB =
+                model.schedulerBuilder("A").build().cast();
+        final InputWire<Integer, Void> bInInteger = taskSchedulerB.buildInputWire("bIn1");
+        final InputWire<Boolean, Void> bInBoolean = taskSchedulerB.buildInputWire("bIn2");
+        final InputWire<String, Void> bInString = taskSchedulerB.buildInputWire("bIn3");
+
+        taskSchedulerA.solderTo(bInInteger);
+        aOutBoolean.solderTo(bInBoolean);
+        aOutString.solderTo(bInString);
+
+        aIn.bind(x -> {
+            if (x % 2 == 0) {
+                aOutBoolean.forward(x % 3 == 0);
+            }
+
+            if (x % 5 == 0) {
+                aOutString.forward(Integer.toString(x));
+            }
+
+            return x;
+        });
```

Comment:
I don't understand this approach for alternative outputs fully. This puts the delegation of tasks inside the business logic. I think a better approach would be to optionally specify the output wire when calling `bind`

Reply:
We discussed this in slack, repeating the conclusion of that discussion here.

I agreed to add this feature on the condition that we first find a use case in our current code or in planned code, and Lazar agreed that this work could be done in a follow up PR.

Reply:
https://github.com/hashgraph/hedera-services/issues/9598